sdk/tools/azure-openrewrite-compiler-maven-plugin/src/main/java/com/azure/openrewrite/goldenimages/GoldenImageCompilerMojo.java

```diff
@@ -0,0 +1,252 @@
+    private static final boolean LIST_CLASSES = false;
+    private static final boolean LOG_COMPILE_COMMAND = false;
+
+    /**
+     * Comma-separated list of profiles to build.
+     * For example: "v1,v2"
+     */
+    @Parameter(property = "targetProfiles", defaultValue = "v1")
+    private String targetProfiles;
+
+    @Parameter(defaultValue = "${project.basedir}", readonly = true)
+    private File baseDir;
+
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
+    @Component
+    private ProjectBuilder projectBuilder;
+
+    public void execute() throws MojoExecutionException {
+        String testResourcesPath = baseDir.getAbsolutePath() + "/src/test/resources/migrationExamples";
+        File testResourcesDir = new File(testResourcesPath);
+
+        if (!testResourcesDir.exists()) {
+            throw new MojoExecutionException("Test resources directory does not exist: " + testResourcesPath);
+        }
+
+        List<File> testFolders = findTestFolders(testResourcesDir);
+        String[] profiles = targetProfiles.split(",");
+
+        for (File folder : testFolders) {
+            for (String profile : profiles) {
+                File profileDir = new File(folder, profile);
+
+                if (!profileDir.exists()) {
+                    getLog().warn("Skipping " + folder.getAbsolutePath() + " as profile directory " + profile + " does not exist.");
+                    continue;
+                }
+
+                File previousProfileDir = getPreviousProfileDir(folder, profile);
+                if (previousProfileDir != null && previousProfileDir.exists()) {
+                    if (hasChanges(profileDir, previousProfileDir)) {
+                        compileJavaFilesWithProfile(profileDir, profile);
+                    } else {
+                        getLog().info("No changes detected between " + profileDir.getAbsolutePath() + " and " + previousProfileDir.getAbsolutePath());
+                    }
+                } else {
+                    compileJavaFilesWithProfile(profileDir, profile);
+                }
+            }
+        }
+    }
+
+    private List<File> findTestFolders(File baseDir) throws MojoExecutionException {
+        try (Stream<Path> paths = Files.walk(baseDir.toPath())) {
+            return paths
+                .filter(Files::isDirectory)
+                .filter(path -> Arrays.stream(targetProfiles.split(","))
+                    .allMatch(profile -> path.resolve(profile).toFile().exists()))
+                .map(Path::toFile)
+                .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new MojoExecutionException("Failed to find test folders", e);
+        }
+    }
+
+    private File getPreviousProfileDir(File folder, String profile) {
```

Comment:
nit: Instead of calling it `previous`, we should consider naming this as base profile to indicate that this is the profile we are using as base.